sdk/include/opentelemetry/sdk/metrics/metric_reader.h

```diff
@@ -72,8 +72,7 @@ class MetricReader
 protected:
 private:
   MetricProducer *metric_producer_;
-  mutable opentelemetry::common::SpinLockMutex lock_;
-  bool shutdown_;
+  std::atomic<bool> shutdown_;
```

Comment:
Please check the comments in PR #2244 for details

```suggestion
  std::atomic<bool> shutdown_{false};
```